src/components/sidebar/ContextAwareSidebar.tsx

```diff
@@ -317,29 +323,16 @@ class ContextAwareSidebar
       </SidebarGroup>
     );
 
-    const getRefGuidFromRef = (ref: Edge) => {
-      const id = stripId(ref.sourceId);
-
-      return Maybe.maybe(course.resourcesById.get(id)).caseOf({
-        just: resource => resource.guid,
-        nothing: () => '',
-      });
-    };
-
     const stripId = (id: string) => {
       const splits = id.split(':');
       if (splits.length === 3) {
         return splits[2];
       }
     };
 
-    const getRefTitleFromRef = (ref: Edge) => {
+    const getRefResourceFromRef = (ref: Edge) => {
       const id = stripId(ref.sourceId);
-
-      return Maybe.maybe(course.resourcesById.get(id)).caseOf({
-        just: resource => resource.title,
-        nothing: () => '[Error loading page title]',
-      });
+      return course.resourcesById.get(id);
```

Comment:
This can still fail if it can't find the resource in the course for whatever reason - we need to either keep it wrapped in a `maybe` and have a default missing title, or, later on in line 357, filter out the undefined resources so that `res.resourceState` does not throw an error

Reply:
filtered out undefined at 357